posixlib/src/main/scala/scala/scalanative/posix/netinet/in.scala

```diff
@@ -40,6 +40,17 @@ object in {
     CUnsignedInt // ipv6mr_interface
   ]
 
+  type ip_mreq = CStruct2[
+    in_addr, // imr_multiaddr
+    in_addr // imr_address
+  ]
+
+  type ip_mreqn = CStruct3[
+    in_addr, // imr_multiaddr
+    in_addr, // imr_address
+    CInt // imr_ifindex
+  ]
+
```

Comment:
As the author & others probably know, through pain &
wasted time,  `ip_mreq` and `ip_mreqn` fall into the
category of "No good home currently defined.

They are _not_ Posix [in.h](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/netinet_in.h.html) so they most definitely do not belong in posixlib `in.scala`.

I found definitions in Linux "man ip" and for Windows. I did
not search for macOs (or FreeBSD).  So, we are left, once
again, with the need for a good home for OS specific declarations. 

Much to my disrepute, I suggest putting these into an
`private [net] ip` object at the bottom of `java.net.SocketHelpers`.   

That way, we can focus on the main issues of this PR and
hide our faces in shame when the next generation of devos asked why we punted.